sql/_34_fig/cbrd_24252/cases/cbrd_24252_34.sql

```diff
@@ -0,0 +1,61 @@
+set trace on;
+
+/* dummy data */
+drop table if exists dummy;
+create table dummy (col_a int);
+insert into dummy
+select rownum from
+table ({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}),
+table ({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}),
+table ({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+
+/* ----------------------------------------
+ *
+ * test case 34
+ *   - child c (parent_col_a:not_null, parent_col_b:not_null)
+ *   - parent p (c:parent_col_a->p:col_a, c:parent_col_b->p:col_b)
+ *
+ * ---------------------------------------- */
+
+drop table if exists t_child;
+drop table if exists t_parent;
+create table t_parent (col_a int, col_b int, c3 int, primary key (col_b, col_a));
+create table t_child (col_a int auto_increment primary key, col_b int);
+alter table t_child add column parent_col_a int not null; /* not null */
+alter table t_child add column parent_col_b int not null; /* not null */
+alter table t_child add constraint foreign key (parent_col_a, parent_col_b) references t_parent (col_a, col_b);
+insert into t_parent select col_a, col_a, col_a from dummy;
+insert into t_child select null, (col_a * -1), col_a, col_a from dummy;
+insert into t_child select null, (col_a * -1), col_a, col_a from dummy;
+
+/* ansi‑style */
+select /*+ recompile */
+    c.col_a,
+    c.col_b
+from
+    t_parent as p
+    inner join t_child as c on c.parent_col_a = p.col_a and c.parent_col_b =  p.col_b
```

Comment:
@youngjinj 
what is the difference with 'cbrd_24252_16.sql' case?

Reply:
The order of the columns that create the primary key and the order of the columns that create the foreign key are different.
1. cbrd_24252_16.sql
   - primary key (col_a, col_b)
   ```
   create table t_parent (col_a int, col_b int, c3 int, primary key (col_a, col_b));
   alter table t_child add constraint foreign key (parent_col_a, parent_col_b) references t_parent (col_a, col_b);
   ```
3. cbrd_24252_34.sql
   - primary key (col_b, col_a)
   ```
   create table t_parent (col_a int, col_b int, c3 int, primary key (col_b, col_a));
   alter table t_child add constraint foreign key (parent_col_a, parent_col_b) references t_parent (col_a, col_b);
   ```

Reply:
Okay thanks for the explanation.